packages/boot/test/integration/controller.booter.app.integration.ts

```diff
@@ -0,0 +1,121 @@
+  };
+
+  // tslint:disable-next-line:no-any
+  let app: any;
+
+  beforeEach(resetSandbox);
+  beforeEach(getApp);
+
+  it('all functions run and work via app.boot() once bound', async () => {
+    const expectedDiscoveredFiles = [
+      `${resolve(SANDBOX_PATH, 'controllers/hello/hello.controller.js')}`,
+      `${resolve(SANDBOX_PATH, 'controllers/two.controller.js')}`,
+      `${resolve(SANDBOX_PATH, 'ctrl/another.ctrl.js')}`,
+    ];
+    const expectedBindings = [
+      `${CoreBindings.CONTROLLERS}.AnotherController`,
+      `${CoreBindings.CONTROLLERS}.HelloController`,
+      `${CoreBindings.CONTROLLERS}.ControllerOne`,
+      `${CoreBindings.CONTROLLERS}.ControllerTwo`,
+    ];
+
+    await app.booter(ControllerBooter);
+    await app.boot(bootOptions);
+    const bootComponent: BootComponent = await app.get(
+      CoreBindings.BOOTCOMPONENT,
+    );
+    const booter: ControllerBooter = await bootComponent.get(
+      `${BootBindings.BOOTER_PREFIX}.ControllerBooter`,
+    );
+
+    // Check Config Phase Ran as expected
+    expect(booter.dirs.sort()).to.eql(bootOptions.controllers.dirs.sort());
+    expect(booter.extensions.sort()).to.eql(
+      bootOptions.controllers.extensions.sort(),
+    );
+    expect(booter.options.nested).to.equal(ControllerDefaults.nested);
+
+    // Check Discovered Phase Ran as expected
+    expect(booter.discovered.sort()).to.eql(expectedDiscoveredFiles.sort());
+
+    // Check Boot Phase Ran as expected
+    const ctrlBindings = app
+      .findByTag(CoreBindings.CONTROLLERS_TAG)
+      .map((b: Binding) => b.key);
+    expect(ctrlBindings.sort()).to.eql(expectedBindings.sort());
+  });
+
+  async function getApp() {
+    await sandbox.copyFile(resolve(__dirname, '../fixtures/application.js'));
+    await sandbox.copyFile(
+      resolve(__dirname, '../fixtures/application.js.map'),
+    );
+    await sandbox.copyFile(
+      resolve(__dirname, '../fixtures/two.controller.js'),
+      'controllers/two.controller.js',
+    );
+    await sandbox.copyFile(
+      resolve(__dirname, '../fixtures/two.controller.js.map'),
+      'controllers/two.controller.js.map',
+    );
+    await sandbox.copyFile(
+      resolve(__dirname, '../fixtures/hello.controller.js'),
+      'controllers/hello/hello.controller.js',
+    );
+    await sandbox.copyFile(
+      resolve(__dirname, '../fixtures/hello.controller.js.map'),
+      'controllers/hello/hello.controller.js.map',
+    );
+    await sandbox.copyFile(
+      resolve(__dirname, '../fixtures/another.ctrl.js'),
+      'ctrl/another.ctrl.js',
+    );
+    await sandbox.copyFile(
+      resolve(__dirname, '../fixtures/another.ctrl.js.map'),
+      'ctrl/another.ctrl.js.map',
+    );
+    await sandbox.copyFile(
+      resolve(__dirname, '../fixtures/empty.controller.js'),
+      'controllers/empty-controller.js',
+    );
+    await sandbox.copyFile(
+      resolve(__dirname, '../fixtures/empty.controller.js.map'),
+      'controllers/empty-controller.js.map',
+    );
```

Comment:
Could you please reduce the number of the fixture files? We don't need to re-check things that have been already verified by unit tests. IMO, a single controller file exporting a single controller class should be enough for this test. In my understanding, we are verifying the integration between the bootstrapper and other moving parts, not implementation details of the bootstrapper itself.